app/views/job_invocations/show.js.erb

```diff
@@ -1,6 +1,22 @@
 $('div#status_chart').html('<%=j job_invocation_chart(@job_invocation) %>');
 $('div#status').flot_pie();
 
+<% if params[:hosts_needs_refresh] == 'true' && @job_invocation.resolved? %>
+  var hosts_tab = $('div#hosts')
+  hosts_tab.html('<%=j render('tab_hosts', :job_invocation => @job_invocation, :hosts => @hosts) %>');
+  hosts_tab.data('needs-refresh', '');
```

Comment:
do you know any better way that does not duplicate the view logic? the only alternative that I can imagine is to render some template html and send just data from server to client, then use the template to clone html and fill in the data but 
  1. we'd duplicate at least translation data or logic like "status 1 means Failed with cross icon"
  2. I saw too many issues with this approach in existing Foreman forms (compute profiles, interface, parameters)

Probably best would be to use some templating language that can be used in both ruby na js (e.g. [mustache](https://mustache.github.io/)) but I don't want to introduce another templating solution :-)